helmfile/values/networkpolicies/common/common.yaml.gotmpl

```diff
@@ -0,0 +1,132 @@
+{{- $netpol := .Values.networkPolicies }}
+
+{{- $apiserver := dict }}
+{{- $ingress := dict }}
+{{- $nodes := dict }}
+
+{{- if eq .Environment.Name "service_cluster" }}
+
+{{- $apiserver = .Values.networkPolicies.global.scApiserver }}
+{{- $ingress = .Values.networkPolicies.global.scIngress }}
+{{- $nodes = .Values.networkPolicies.global.scNodes }}
+
+{{- else if eq .Environment.Name "workload_cluster" }}
+
+{{- $apiserver = .Values.networkPolicies.global.wcApiserver }}
+{{- $ingress = .Values.networkPolicies.global.wcIngress }}
+{{- $nodes = .Values.networkPolicies.global.wcNodes }}
+
+{{- end -}}
+
+rules:
+  egress-rule-dns:
+    peers:
+      - cidr: 10.233.0.3/32
+    ports:
+      - tcp: 53
+      - udp: 53
+
+  egress-rule-apiserver:
+    peers:
+      {{- range $apiserver.ips }}
+      - cidr: {{ . }}
+      {{- end }}
+    ports:
+      - tcp: {{ $apiserver.port }}
+
+  egress-rule-ingress:
+    peers:
+      {{- if and $netpol.global.externalLoadBalancer $ingress.ips }}
+      {{- range $ingress.ips }}
+      - cidr: {{ . }}
+      {{- end }}
+      {{- else if and $netpol.global.ingressUsingHostNetwork $nodes.ips }}
+      {{- range $nodes.ips }}
+      - cidr: {{ . }}
+      {{- end }}
+      {{- else }}
+      {{- range $ingress.ips }}
+      - cidr: {{ . }}
+      {{- end }}
```

Comment:
```suggestion
      {{- if and $netpol.global.ingressUsingHostNetwork $nodes.ips }}
      {{- range $nodes.ips }}
      - cidr: {{ . }}
      {{- end }}
      {{- else }}
      {{- range $ingress.ips }}
      - cidr: {{ . }}
      {{- end }}
```

Reply:
This doesn't really work as we can have ingress through host ports even with an external load balancer, and the latter needs to take priority in the egress rule.

I'm trying to find another way to do it but I'm not finding a good solution. :thinking: